src/terminal/parser/stateMachine.cpp

```diff
@@ -353,6 +420,13 @@ void StateMachine::_ActionEscDispatch(const wchar_t wch)
 {
     _trace.TraceOnAction(L"EscDispatch");
 
+    if (_isStringTerminatorIndicator(wch))
+    {
+        // Presumably, ST can be safely ignored.
+        _trace.DispatchSequenceTrace(true);
+        return;
+    }

```

Comment:
I don't think this is the right place to ignore an ST sequence. The state machine is shared by both the input and output engines, and in the case of the input engine, an ST escape represents the key combination <kbd>Alt</kbd>+<kbd>\\</kbd>. So you probably need to let this be passed through to the `ActionEscDispatch`, and then add a case for it in the `OutputStateMachineEngine` that just returns true.

Reply:
That was my initial thought, to add a case in `OutputStateMachineEngine::ActionEscDispatch`. But `ESC \` does not seem to belong to the ASCII escape code category. I don't know whether `\` is a special case that's widely known or it's something wrong about the current state machine implementation. Point is, as a VT newbie I'm trying to do the right thing but adding a special case in `OutputStateMachineEngine::ActionEscDispatch` does not seem right enough for me.

Reply:
It's a legitimate escape sequence - it matches the pattern of ESC followed by a final char in the range 3/0 to 7/14. It's just that it's a noop in general, but indicates the end of string when encountered within a string sequence. It's only a complication for us because we've got this thing where we "pass through" any sequence that isn't explicitly handled - otherwise it could just be ignored silently and wouldn't be a problem.

If you don't like the current approach though, the other option would be to add a check to see which state machine engine is being used, and only ignore it for output. You can see the SS3 code doing something similar with a `ParseControlSequenceAfterSs3` check, but that's a bit of a pain to implement. Ideally I'd like us to get rid of all those checking functions and just have a simple property indicating input or output, but that's probably not something to get into now.